int32 ID
bool completed
string hash
bool received
int32 from_robot
